### ISSUE TITLE
Extendable block names in child templates, e.g. name:suffix

### ISSUE THREAD

**Issue by TiGR** [twigphp/Twig#1488]
TLDR: provide ability to extend parent template block names with suffixes, e.g. "name:suffix", thus providing more flexibility for inheritance. This is especially useful for horizontal reuse. The behavior should be exactly the same as if there was suffixed block in parent element located at the same place as unsuffixed one.

When you inherit a template and want to add something to some block, you have to use parent() in order to keep content from ancestors. And in order to just append data new content to block, without overriding it you can't just add parent() to all blocks, because you have to make sure that block exists in some of parents, and all this may become a problem, especially if you use horizontal reuse. Moreover, you cant override content from specific ancestor, but keep others, you can only override entire block.

Here is a feature request that might fix all these problems: make block names extendable. Say, in main layout file you have a block named 'scripts'. You can add names like 'scripts:feature_one' and 'scripts:another_feature'. These get overriden only if you specificy their exact name. In other cases they stay intact.

You can achiveve the same thing by creating blocks 'scripts_feature_one' and 'scripts_another_feature' in main twig file, but then you have to initialize blocks for all possible extensions in your project.

And if there if some block is not used in parent it would be ignored silently.

**Comment by fabpot** [comment]
Closing this old feature request as it's not simple to implement and I fear it would increase both code complexity on Twig's side and the learning curve for our users.
